vm/src/vm.rs

```diff
@@ -407,10 +407,23 @@ impl VirtualMachine {
     }
 
     pub fn import(&self, module: &str, from_list: &PyObjectRef, level: usize) -> PyResult {
-        let sys_modules = self.get_attribute(self.sys_module.clone(), "modules")?;
-        sys_modules
-            .get_item(module.to_string(), self)
-            .or_else(|_| {
+        // if the import inputs seem weird, e.g a package import or something, rather than just
+        // a straight `import ident`
+        let weird = module.contains('.')
+            || level != 0
+            || objbool::boolval(self, from_list.clone()).unwrap_or(true);
+
+        let module = self.new_str(module.to_owned());
+
+        let sys_module = if weird {
```

Comment:
This variable name should be renamed. It is not the `sys` module, but it is optionally a cached module from `sys.modules`.